### PR TITLE
Make `:javap` in REPL work on JDK 11+

### PR DESCRIPTION
**Pull request by som-snytt**

### CODE REVIEW

File: src/repl-frontend/scala/tools/nsc/interpreter/shell/JavapClass.scala [hunk 1]
```diff
@@ -10,55 +10,53 @@
...
+  tool: JavapTool
+) {
   import JavapClass._
+  import Javap.{DefaultOptions, HashSplit, helper, toolArgs}
+  import JavapTool.Input
+  import java.io.FileNotFoundException
+  import scala.reflect.io.File
 
-  lazy val tool = JavapTool()
+  //private val printWriter: PrintWriter = new IMain.ReplStrippingWriter(intp)
```
Comment (som-snytt) [inline src/repl-frontend/scala/tools/nsc/interpreter/shell/JavapClass.scala]: Distracted edit, we were watching Batwoman.

File: src/repl/scala/tools/nsc/interpreter/Interface.scala [hunk 2]
```diff
@@ -193,6 +194,10 @@ trait Repl extends ReplCore {
   // like beQuietDuring, but also turn off noisy settings.
   // this requires access to both settings and the global compiler
   def withSuppressedSettings(body: => Unit): Unit
+
+  def compilerClasspath: Seq[URL]
+
+  def outputDir: AbstractFile
```
Comment (som-snytt) [inline src/repl/scala/tools/nsc/interpreter/Interface.scala:200]: suboptimal for now, bytecode printing should be a service.

**Comment by som-snytt** [comment]
Requested retronym who requested a less budget-busting version.

The LOC delta is a wash after refactor. I've already forgotten what it does.

**Review by retronym — APPROVED** [review]
Thanks for bravely straddling the JDK version chasm.

Later, we could consider making the code cleaner by making our build messier letting sub-projects be built with particular JDKs (or JDK.latest with particular `-release ` arguments) and packaging the result up in a multi-release JAR.  This could be useful in other spots of the library.

**Comment by SethTisue** [comment]
and another "Forward port javap support" ticket: https://github.com/lampepfl/dotty-feature-requests/issues/80

**Pull request merged**